### PR TITLE
added the interrupt based mechanism

### LINKED ISSUE #121298: Why eventf API of go-client requires polling
Hello All,

I am referring this code and line
https://github.com/kubernetes/kubernetes/blob/master/test/integration/events/events_test.go#L76

First I would like to thank to author for adding this example. This piece of code is really helpful to develop more complicated solution based on k8s `go-client`. But I have few questions, which arouses when we go for the more complicated event posting mechanism.  

- One question, why polling? In a limited resource environment, when we try to send multiple events, and poll for every request, the static value 100 and 20 might not be effective, considering the load on api-server. Persistence of the sent event might take longer. Now we can argue some 99 percentile would follow this pattern, but that is not the answer. This is non-deterministic and theoretically the timeout can happen whatever the value we set. 
So my question is, **Can we anyhow modify the example so that instead of polling, some `interrupt` based mechanism would notify the caller, event has been handled to api-server for processing**. 

Thanks in advance.

### PR DESCRIPTION
**Pull request by swastik959**
#### What type of PR is this?
/cleanup

#### What this PR does / why we need it:
replaces deprecated pooling based mechanism with the interrupt based one 
#### Which issue(s) this PR fixes:

Fixes #121298 

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by swastik959** [comment]
@dims can you please review it please

**Comment by cblecker** [comment]
@swastik959 please remove the unneeded modifications to the go.sum file. Thanks!

**Comment by swastik959** [comment]
@cblecker actually it was modified when I ran go mod tidy

**Comment by cblecker** [comment]
@swastik959 then there is likely an issue with your local environment. That file is forced to be conformant in the repo, and will fail tests if it doesn't match what CI thinks it should be

/ok-to-test

File: test/integration/events/events_test.go [hunk 1]
```diff
@@ -73,28 +73,32 @@ func TestEventCompatibility(t *testing.T) {
 	newRecorder := newBroadcaster.NewRecorder(scheme.Scheme, "k8s.io/kube-scheduler")
 	newBroadcaster.StartRecordingToSink(stopCh)
 	newRecorder.Eventf(regarding, related, v1.EventTypeNormal, "memoryPressure", "killed", "memory pressure")
-	err = wait.PollImmediate(100*time.Millisecond, 20*time.Second, func() (done bool, err error) {
```
Comment (cblecker) [inline test/integration/events/events_test.go:76]: We should move this to https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollUntilContextTimeout as opposed to rewriting it to not use the apimachinery

Comment (swastik959) [reply test/integration/events/events_test.go:76]: actually the issue was to replace the polling based mechanism to a  interrupt based mechanism

**Pull request closed without merging**